src/dmd/parse.d

```diff
@@ -5018,7 +5018,10 @@ class Parser(AST) : Lexer
                     continue;
 
                 default:
-                    error("semicolon expected, not `%s`", token.toChars());
+                    if (loc.linnum != token.loc.linnum)
+                        error("semicolon needed to end declaration of `%s` begun on line %d, instead of `%s`", v.toChars(), loc.linnum, token.toChars());
```

Comment:
Add an errorSupplemental (when verrors=context?) because having line information packed into an error is just awkward when you instead could have a new error line saying that the statement began here (with context).

Reply:
Ping @WalterBright ! We should really try to get this in.